composer/utils/inference.py

```diff
@@ -148,16 +160,15 @@ def export_for_inference(
             export_model = None
             try:
                 export_model = torch.jit.script(model)
-            except Exception as e:
-                log.warning(
-                    'Scripting with torch.jit.script failed with the following exception. Trying torch.jit.trace!',
-                    exc_info=True)
+            except Exception:
                 if sample_input is not None:
+                    log.warning('Scripting with torch.jit.script failed. Trying torch.jit.trace!',)
                     export_model = torch.jit.trace(model, sample_input)
                 else:
-                    raise RuntimeError(
-                        'Scripting with torch.jit.script failed and sample inputs are not provided for tracing with torch.jit.trace'
-                    ) from e
+                    log.warning(
+                        'Scripting with torch.jit.script failed and sample inputs are not provided for tracing '
+                        'with torch.jit.trace',
+                        exc_info=True)
 
             if export_model is not None:
```

Comment:
Should we raise an exception in case `export_model` is None, to avoid silent failure?

Reply:
Done. However, jit or script would have already raised an error.